### PR TITLE
Code dump

### PR DESCRIPTION
**Pull request by lavalamp**
Now that there's a place for this to go, I'm going to send some PRs here instead of to the main repo.

### CODE REVIEW

**Comment by apelisse** [comment]
`Code dump`. I can't stop reading `core dump` :-)

Anyway, has this already been reviewed or should we review it?

**Comment by lavalamp** [comment]
I think this code has mostly not been reviewed, so at least read it--I will probably want to fix things in followups rather than hold it up, though.

File: value/value.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+// Map is a map of key-value pairs. It represents both structs and maps. We use
+// a list and a go-language map to preserve order.
+//
+// Set and Get helpers are provided.
+type Map struct {
+	Items []Field
+
+	// may be nil; lazily constructed.
+	// TODO: Direct modifications to Items above will cause serious problems.
```
[the remark this answers is not in the document]
Comment (lavalamp) [reply value/value.go]: I want to see how the usage evolves first. Maybe I can even combine the Map and List types if I'm clever.

File: value/value.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+// BooleanValue returns b as a scalar boolean Value.
+func BooleanValue(b bool) Value {
+	b2 := Boolean(b)
+	return Value{Boolean: &b2}
+}
+
+// HumanReadable returns a human-readable representation of the value.
+// TODO: Rename this to "String".
```
[the remark this answers is not in the document]
Comment (lavalamp) [reply value/value.go]: Yeah the problem is the name is already in use for a good cause :/

File: testvectors/test.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	LastObject YAMLObject `yaml:"lastObject"`
+	LiveObject YAMLObject `yaml:"liveObject"`
+	NewObject  YAMLObject `yaml:"newObject"`
+
+	ExpectedObject    YAMLObject    `yaml:"expectedObject"`
+	ExpectedConflicts []interface{} `yaml:"expectedConflicts"` // TODO: get correct type
+
+	// TODO: do we need to indicate whether this supposed to be
+	// controller-desired or user-desired behavior? Or is it sufficient for
+	// controllers to just not specify the LastObject?
```
[the remark this answers is not in the document]
Comment (lavalamp) [reply testvectors/test.go]: Yes, that's stuff to consider. I think the TODO is clear enough for now.

File: testvectors/test.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+// Schemas keeps the schemas that may be referenced by the test vectors.
+var Schemas = map[string]SchemaDefinition{}
+
+// Implementation is a stand-in for an implementation. It's structured so that
+// you can run the same vectors through a system that does the three-way merge
+// directly, or through a system that does a separate diff and merge.
+type Implementation interface {
+	// Test must not modify v. Test should call t.Parallel().
```
[the remark this answers is not in the document]
Comment (lavalamp) [reply testvectors/test.go]: Will fix

File: value/unstructured.go [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+		return Value{Float: &f}, nil
+	case float64:
+		f := Float(t)
+		return Value{Float: &f}, nil
+	case string:
+		return StringValue(t), nil
+	case bool:
+		return BooleanValue(t), nil
+	}
+}
```
[the remark this answers is not in the document]
Comment (lavalamp) [reply value/unstructured.go]: Will fix

**Pull request merged**